From f0ce70989526fc9a0223398c99ea0d09777ea5df Mon Sep 17 00:00:00 2001
From: Martin Thomson <martin.thomson@gmail.com>
Date: Thu, 15 Feb 2018 16:34:02 +1100
Subject: [PATCH] Bug 1438426 - Avoid stringop-truncation warning, r=franziskus

--HG--
extra : rebase_source : 4ea1630d0da0ce3523309e3da33ee50961682242

Upstream-commit: https://github.com/nss-dev/nss/commit/f0ce70989526fc9a0223398c99ea0d09777ea5df
[Thomas: edited after git format-patch to add the nss/ prefix needed
for the patch to apply properly on the source code extracted by the
tarball.]
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
---
 nss/coreconf/nsinstall/pathsub.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nss/coreconf/nsinstall/pathsub.c b/nss/coreconf/nsinstall/pathsub.c
index a42a9f30a..c31a946f0 100644
--- a/nss/coreconf/nsinstall/pathsub.c
+++ b/nss/coreconf/nsinstall/pathsub.c
@@ -212,7 +212,7 @@ reversepath(char *inpath, char *name, int len, char *outpath)
 	    xchdir("..");
 	} else {
 	    cp -= 3;
-	    strncpy(cp, "../", 3);
+	    memcpy(cp, "../", 3);
 	    xchdir(buf);
 	}
     }
-- 
2.14.3

